src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/esql_state_manager.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { StateComparators } from '@kbn/presentation-publishing/state_manager';
+import { initializeStateManager } from '@kbn/presentation-publishing/state_manager';
+import type { PublishingSubject } from '@kbn/presentation-publishing';
+import type { ESQLControlVariable, ESQLVariableType, EsqlControlType } from '@kbn/esql-types';
+import type {
+  OptionsListESQLControlState,
+  OptionsListSelection,
+} from '../../../../common/options_list';
+
+type ESQLOptionsListInputState = Partial<
+  Pick<OptionsListESQLControlState, 'esqlQuery' | 'variableName' | 'variableType' | 'controlType'>
```

Comment:
We don't pull `availableOptions` into the state here, the options list already has an `availableOptions` state and those get derived within `fetchAndValidate`.

The follow-up PR will apply a static values control's `availableOptions` instead of fetching data.